getsentry/sentry
# Maintain Query Filters

#### What's this PR do?

This refactors the Stream Filters quite a bit so that state can be tracked and updated appropriately. The query state and filters are now moved into Stream and passed down to the other components. We also broke out the child components into new files.

Some changes you can see in the behavior:
- Clearing the search box works and is independent of the filters
- Clicking the "Stream" link at the top goes back to the default state
- When the filters and search box match the default query, the URL becomes pretty again with no query params

This fixes "Maintain query and filters when toggling bookmarks/assigned" on issue #1585.
#### Where should the reviewer start?

`src/sentry/static/sentry/app/views/stream.jsx`
#### How should this be manually tested?

Run `make test-js`

You can also start the Sentry server, navigate to the stream page, and play around with the searching and filtering. Also try clicking the "clear search" button and clicking the "Stream" link in the menu bar to see how that behaves.
#### Any background context you want to provide?

Previously the child components were tracking their own state and this caused some syncing problems. Moving the state logic into Stream helps to keep everything in order between the UI, URL, and current query.
#### What gif best describes this PR or how it makes you feel?

![aaawwWWWWWW](http://i.giphy.com/wQGD7YKEBNsWY.gif)
